eden/RNA.py

```diff
@@ -179,8 +180,70 @@ def _align_sequence_structure(self, seq, neighs):
                 clean_seq, clean_struct = rnafold.RNAfold_wrapper(seq[1])
             else:
                 clean_seq, clean_struct = make_seq_struct(seed, struct)
+            if structure_deletions:
+                clean_struct = self._clean_structure(clean_seq, clean_struct)
+
         return header, clean_seq, clean_struct, energy
 
+    def _clean_structure(self, seq, stru):
+        '''
+        Parameters
+        ----------
+        seq : basestring
+            rna sequence
+        stru : basestring
+            dotbracket string
+
+        Returns
+        -------
+        the structure given may not respect deletions in the sequence.
+        we transform the structure to one that does
+        '''
+
+        # find  deletions in sequence
+        ids = []
+        for i, c in enumerate(seq):
+            if c == '-':
+                ids.append(i)
+        # remove brackets that dont have a partner anymore
+        stru = list(stru)
+        pairdict = self._pairs(stru)
+        for i in ids:
+            stru[pairdict[i]] = '.'
+        # delete deletions in structure
+        ids.reverse()
+        for i in ids:
+            del stru[i]
+        stru = ''.join(stru)
+
+        # removing obvious mistakes
+        stru = stru.replace("(())", "....")
```

Comment:
I'm not a huge fan of adding a new dependency to EDeN but this looks like a huge hack and pyparsing would probably a good choice for all RNA-structure handling.
ping @fabriziocosta